src/groups/bmq/bmqc/bmqc_multiqueuethreadpool.h

```diff
@@ -993,74 +794,9 @@ inline void MultiQueueThreadPool<TYPE>::processQueue(int queue)
         }
 
         d_config.d_eventCallbackFn(queue, info.d_context_p.get(), event);
-
-        if (unrefEvent(event, false)) {
-            bool finalize = (d_config.d_finalizeType ==
-                             Config::BMQC_FINALIZE_ALL);
-            finalize      = finalize || ((d_config.d_finalizeType ==
-                                     Config::BMQC_FINALIZE_MULTI_QUEUE) &&
-                                    event->d_enqueuedOnMultipleQueues);
-
-            if (finalize) {
-                event->d_type = Event::BMQC_FINALIZE_EVENT;
-                d_config.d_eventCallbackFn(queue,
-                                           info.d_context_p.get(),
-                                           event);
-            }
-            d_pool.releaseObject(event);
-        }
     }
 }
 
-template <typename TYPE>
-inline int MultiQueueThreadPool<TYPE>::enqueueEventImp(Event* event, int queue)
-{
-    // PRECONDITIONS
-    BSLS_ASSERT_SAFE(event);
-    BSLS_ASSERT_SAFE(isStarted() && "MQTP has not been started");
-
-    /// Value for the various RC error categories
-    enum RcEnum {
-        /// No error
-        rc_SUCCESS = 0,
-
-        /// An error was encountered while pushing back (queue is disabled)
-        rc_PUSH_BACK_ERROR = -1
-    };
-
-    // Determine start and end queues for which to enqueue
-    int startQueue = queue;
-    int endQueue   = queue + 1;
```

Comment:
Don't want to pay for these operations when we want to enqueue to just 1 queue, simplified